scripts/nfd-must-gather.sh

```
@@ -0,0 +1,393 @@
+
+    # This function is now called from collect_nfd_operator for each namespace
+    # But we'll also collect from the default NFD namespace if it wasn't already covered
+    if oc get namespace "${NFD_NAMESPACE}" &> /dev/null; then
+        collect_logs_for_namespace "${NFD_NAMESPACE}"
+    fi
+
+    # Collect any NFD-related pods from other namespaces that might have been missed
+    log_info "Scanning all namespaces for additional NFD pods"
+    while IFS= read -r line; do
+        if [[ -n "$line" && "$line" != "NAMESPACE"* ]]; then
+            local ns=$(echo "$line" | awk '{print $1}')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)